### ISSUE TITLE
source directive and --source-path

### ISSUE THREAD

**Issue by virgilwashere** [koalaman/shellcheck#1577]
# [SC1090] and [-P/--source-path SCRIPTDIR][hash]

#### For new checks and feature suggestions
- [X] I am using the latest daily build

- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything ~related~ **definitive**

- [X] The [rule's wiki page][SC1090] does not already cover this, and I've added a reference to [-P/--source-path SCRIPTDIR][hash]

[SC1090]: <https://shellcheck.net/wiki/SC1090>
[hash]:<https://github.com/koalaman/shellcheck/commit/301705edea829c5153c43bf52215d09b5e00d182>

✏️ @koalaman, in #882, you wrote:
> With [301705e][hash] you can specify `-P SCRIPTDIR` to search relative to the script's directory, or `-P /my/path` to search relative to a path in the filesystem.
> 
> This affects both relative and absolute paths, both in `source` directives and literal pathnames handed to `source`/`.`

#### ➡️ `source` directive:

```bash
$ ls ${LIBDIR}
goodstuff.bash tryinghard.sh
$ cat ${LIBDIR}/tryinghard.sh
#!/bin/bash
# shellcheck source=SCRIPTDIR
source ${LIBDIR}/goodstuff.bash
```

#### 🔀 `source-path` directive:

```bash
#!/bin/bash
# shellcheck source-path=SCRIPTDIR
source ${LIBDIR}/goodstuff.bash
```

#### ⚙️ `.shellcheckrc` option

```console
$ cat .shellcheckrc
source-path=SCRIPTDIR:./libs
$
```

**Comment by koalaman** [comment]
This should definitely be a thing, and now it is: you can use `source-path` directives in files and `.shellcheckrc`s

**Comment by YorkZ** [comment]
None of the methods work for me. I'm using `shellcheck` v0.7.1 on Arch Linux. My `.shellcheckrc` file is:

```
source-path=SCRIPTDIR
```
I've tried placing this file in both my home directory, and the directory of my script, none of them worked.

I then tried adding the following line in my script:
```
# shellcheck source-path=SCRIPTDIR
```
so that my script looks like:
```
this_file=$(readlink -f "${BASH_SOURCE[0]}")
this_dir=$(dirname "${this_file}")
# shellcheck source-path=SCRIPTDIR
source "${this_dir}/menu-common.sh"
```
still doesn't work.

Finally, I tried:
```
shellcheck --source-path=SCRIPTDIR customer-menu
```
which doesn't work.

**Comment by brother** [comment]
@YorkZ probably better follow up in #1998 - this issue was closed when it worked and something has happened during the last year or so...
